### ISSUE TITLE
Feature Suggestion: Detect redundant entries in `.rubocop_todo.yml`

### ISSUE THREAD

**Issue by andyw8** [rubocop-hq/rubocop#8019]
## Is your feature request related to a problem? Please describe.

When adding RuboCop to an existing codebase, it's common to add an automatically generated `.rubocop_todo.yml`. Over time, this can become stale as code is changed. Entries in the config can end up referring to issues which no longer exist. The additional noise makes it harder to tell how many genuine issues are remaining.

## Describe the solution you'd like

Add a flag to detect if the config includes redundant entries. If so, it should suggest that the user re-generates `.rubocop_todo.yml`.

The implementation would likely involve re-running RuboCop with the `--auto-gen-config` and comparing it to the existing one, ignoring any comments.

## Describe alternatives you've considered

n/a

## Additional context

n/a

**Comment by Mangara** [comment]
This would be really useful!

An alternative implementation could be to run the cop as normal for excluded files, but to report an error if the cop passes (or if the file no longer exists). Glob patterns may need special handling as well.

**Comment by jonas054** [comment]
A problem I see is that sometimes we exclude files or disable cops because we're not interested in the results of that inspection. It's not always to hide existing offenses. This would be a problem if we want to implement checking of configuration in `./.rubocop.yml`, or similar.

For `.rubocop_todo.yml` it's a bit more straight forward. The exclusions and disablings made there could always be checked to see if they are still relevant. On the other hand, is that really a significant improvement over re-running `rubocop --auto-gen-config` and checking the diff to see if you want to keep the changes in `.rubocop_todo.yml`?

**Comment by andyw8** [comment]
> For .rubocop_todo.yml it's a bit more straight forward. 

Yes, it would probably make sense for this to default to only checking `.rubocop_todo.yml`, perhaps with an option to override with another filename.

> On the other hand, is that really a significant improvement over re-running rubocop --auto-gen-config

It's not game-changing, it's would just one less thing to have to remember to do manually.
